workspaces/src/rpc/query.rs

```diff
@@ -0,0 +1,393 @@
+            _ => Err(RpcErrorCode::QueryReturnedInvalidData.message("while querying state")),
+        }
+    }
+}
+
+impl<'a> Query<'a, ViewState> {
+    pub(crate) fn view_state(client: &'a Client, id: &AccountId) -> Self {
+        Self::new(
+            client,
+            ViewState {
+                account_id: id.clone(),
+                prefix: None,
+            },
+        )
+    }
+
+    /// Set the prefix for viewing the state.
+    pub fn prefix(mut self, value: &[u8]) -> Self {
+        self.method.prefix = Some(value.into());
+        self
+    }
+}
+
+impl ProcessQuery for ViewAccessKey {
+    type Method = methods::query::RpcQueryRequest;
+    type Output = AccessKey;
+
+    fn into_request(self, block_reference: BlockReference) -> Result<Self::Method> {
+        Ok(Self::Method {
+            block_reference,
+            request: QueryRequest::ViewAccessKey {
+                account_id: self.account_id,
+                public_key: self.public_key.into(),
+            },
+        })
+    }
+
+    fn from_response(resp: <Self::Method as RpcMethod>::Response) -> Result<Self::Output> {
+        match resp.kind {
+            QueryResponseKind::AccessKey(key) => Ok(key.into()),
+            _ => Err(RpcErrorCode::QueryReturnedInvalidData.message("while querying access key")),
+        }
+    }
+}
+
+impl ProcessQuery for ViewAccessKeyList {
+    type Method = methods::query::RpcQueryRequest;
+    type Output = Vec<AccessKeyInfo>;
+
+    fn into_request(self, block_reference: BlockReference) -> Result<Self::Method> {
+        Ok(Self::Method {
+            block_reference,
+            request: QueryRequest::ViewAccessKeyList {
+                account_id: self.account_id,
+            },
+        })
+    }
+
+    fn from_response(resp: <Self::Method as RpcMethod>::Response) -> Result<Self::Output> {
+        match resp.kind {
+            QueryResponseKind::AccessKeyList(keylist) => {
+                Ok(keylist.keys.into_iter().map(Into::into).collect())
+            }
+            _ => Err(RpcErrorCode::QueryReturnedInvalidData.message("while querying access keys")),
+        }
+    }
+}
+
+impl ProcessQuery for GasPrice {
+    type Method = methods::gas_price::RpcGasPriceRequest;
+    type Output = Balance;
+
+    fn into_request(self, block_ref: BlockReference) -> Result<Self::Method> {
+        let block_id = match block_ref {
+            // User provided input via `block_hash` or `block_height` functions.
+            BlockReference::BlockId(block_id) => Some(block_id),
+            // default case, set by `Query` struct via BlockReference::latest.
+            BlockReference::Finality(_finality) => None,
+            // Should not be reachable, unless code got changed.
+            other => {
+                return Err(RpcErrorCode::QueryFailure.message(format!(
+                    "Invalid block reference provided to gas price: {other:?}"
+                )))
+            }
```

Comment:
nit: would be nice if the other case is handled specifically, so that if another variant is added in the future, it won't default to erroring